### DISCUSSION TITLE
Invalid unicode error when reading CSV file - How to handle?

### DISCUSSION THREAD

**Discussion by tbeason** [duckdb/duckdb#7291] [Q&A]
I get an error `Invalid unicode (byte sequence mismatch) detected in CSV file.` when reading a particular CSV file. I had thought that there might be a way to use the `types` option in the CSV parser to get around the problem, but I have not had success. I do not need the particular column in which the error appears, but there are almost 100 columns, so I do not think manually typing them into the `columns` argument of the CSV parser is feasible. I see there is an `ignore_errors` option, but (1) it does not actually fix the problem and (2) I do not want to skip the entire row, just that particular column.

Any suggestions?

**Comment by Alex-Monahan** [comment]
Howdy! I've worked around that problem by converting the file to UTF-8 in Python using a library like these: https://dev.to/bowmanjd/character-encodings-and-detection-with-python-chardet-and-cchardet-4hj7

Then DuckDB ahould read it!

**Reply by rsund** [reply]
Conversion to UTF-8 with external tool works, but it is a bit clumsy to do that for 100GB+ files just because of a couple invalid characters. So I agree that an option to force invalid unicode to some valid character would be really useful feature in such cases.

**Reply by tbeason** [reply]
Yea. Of course I can use some other tool to work with it. For example, Julia's CSV.jl just works. No additional special options or anything required. But I would like if there were some option in DuckDB.
